cvmfs/server/cvmfs_server_mkfs.sh

```diff
@@ -325,22 +331,39 @@ cvmfs_server_mkfs() {
     echo -n "(repository flagged volatile)... "
   fi
   local user_shell="$(get_user_shell $name)"
-  local create_cmd="$(__swissknife_cmd) create  \
-    -t $temp_dir                                \
-    -r $upstream                                \
-    -n $name                                    \
-    -a $hash_algo $volatile_opt                 \
-    -o ${temp_dir}/new_manifest                 \
-    -R $(get_reflog_checksum $name)"
-  if $garbage_collectable; then
-    create_cmd="$create_cmd -z"
-  fi
-  if [ "x$voms_authz" != "x" ]; then
-    echo -n "(repository will be accessible with VOMS credentials $voms_authz)... "
-    create_cmd="$create_cmd -V $voms_authz"
+  if [ x"$upstream_type" != xgw ]; then
+      local create_cmd="$(__swissknife_cmd) create  \
+      -t $temp_dir                                \
+      -r $upstream                                \
+      -n $name                                    \
+      -a $hash_algo $volatile_opt                 \
+      -o ${temp_dir}/new_manifest                 \
+      -R $(get_reflog_checksum $name)"
+      if $garbage_collectable; then
+          create_cmd="$create_cmd -z"
+      fi
+      if [ "x$voms_authz" != "x" ]; then
+          echo -n "(repository will be accessible with VOMS credentials $voms_authz)... "
+          create_cmd="$create_cmd -V $voms_authz"
+      fi
+
+      $user_shell "$create_cmd" > /dev/null                       || die "fail! (cannot init repo)"
+      sign_manifest $name ${temp_dir}/new_manifest $repoinfo_file || die "fail! (cannot sign repo)"
+  else
```

Comment:
The entire else-block can go away, no?  The create command is anyway not executed.